content/guides/contribute/handling-new-patterns.mdx

```diff
@@ -0,0 +1,60 @@
+---
+title: Handling new patterns
+description: Guidance to help Primer consumers decide what to do when discovering a new design pattern. This primary audience for this guidance is GitHub staff.
+---
+
+We know that Primer isn't going to always have exactly what you want when designing a new feature. Explorations done by our product designers are what help us innovate and create components to solve problems that the Primer maintainers haven't considered yet. The Primer maintainers may incorporate those ideas by balancing that work with our bigger strategic initiatives to mature Primer.
+
+## Exploring new components that aren't ready for sharing
+
+We strongly prefer design patterns that are consistent and can be reused for other features, but there is room for patterns that aren't meant to be reused. Our goal isn't to only build a UI from existing components, it's to help all users complete tasks in the most efficient and pleasant way possible. As long as what you ship is accessible, visually consistent, and high-quality: go for it.
+
+Signs that your new component isn't ready to be shared outside of your app:
+
+- The component was rushed and has known issues that need to be addressed after launch
+- It solves a very specific design problem my team encountered, but is unlikely to be useful for other use-cases
+- It's a very complex component that applies to a use-case other apps don't often deal with
+
+## Sharing new components before they land in Primer
+
+GitHub's designers come up with innovative ways to solve difficult problems. Our teams and our products will be stronger if we make our good work available for others, even if it's not through Primer.
+
+There's a long process to vet something into Primer, and we understand how that doesn't always work with your team's deadlines to ship features. Instead of trying to add your new pattern directly into Primer you can start by focusing on how it works for the use-cases your team identified and share it for other designers to try out in their work.
+
+Signs that your component is a good candidate for sharing:
+
+- It's very likely to be used in more than one app
```

Comment:
```suggestion
- It's very likely to be used in more than one product
```

 I think we refer to them as products. Apps is used for pointing to our Android and iOS app?

Reply:
I've heard engineers refer to them as "apps", but I see the confusion with iOS and Android. I'm ok with changing it to "products" if other reviewers share this concern.